XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs

```diff
@@ -0,0 +1,960 @@
+							{
+								initTaskSource?.TrySetResult(device);
+								Element?.RaiseMediaCaptureFailed($"Camera device error: {error}");
+								CloseDevice(device);
+							},
+							OnClosedAction = device =>
+							{
+								initTaskSource?.TrySetResult(null);
+								CloseDevice(device);
+							}
+						},
+						backgroundHandler);
+
+					captureSessionOpenCloseLock.Release();
+					device = await initTaskSource.Task;
+					initTaskSource = null;
+					if (device != null)
+					{
+						await PrepareSession();
+					}
+				}
+				catch (Java.Lang.Exception error)
+				{
+					LogError("Failed to open camera", error);
+					Available = false;
+				}
+				finally
+				{
+					IsBusy = false;
+				}
+			}
+		}
+
+		public void UpdateCaptureOptions()
+		{
+			switch (Element.CaptureOptions)
+			{
+				default:
+				case CameraCaptureOptions.Photo:
+					cameraTemplate = CameraTemplate.Preview;
+					break;
+				case CameraCaptureOptions.Video:
+					cameraTemplate = CameraTemplate.Record;
+					break;
+			}
+		}
+
+		public void TakePhoto()
+		{
+			if (IsBusy || cameraTemplate != CameraTemplate.Preview)
+				return;
+
+			try
+			{
+				if (device != null)
+				{
+					session.StopRepeating();
+					repeatingIsRunning = false;
+					sessionBuilder.AddTarget(photoReader.Surface);
+					sessionBuilder.Set(CaptureRequest.FlashMode, (int)flashMode);
+					session.Capture(sessionBuilder.Build(), null, null);
+					sessionBuilder.RemoveTarget(photoReader.Surface);
+					UpdateRepeatingRequest();
+				}
+			}
+			catch (Java.Lang.Exception error)
+			{
+				LogError("Failed to take photo", error);
+			}
+		}
+
+		void OnPhoto(object sender, byte[] data) =>
+			Device.BeginInvokeOnMainThread(() =>
+			{
+				Element?.RaiseMediaCaptured(new MediaCapturedEventArgs(data, ImageSource.FromStream(() => new MemoryStream(data))));
+			});
+
+		void OnVideo(object sender, string data) => 
+			Device.BeginInvokeOnMainThread(() =>
+			{
+				Element?.RaiseMediaCaptured(new MediaCapturedEventArgs(video: MediaSource.FromFile(data)));
+			});
+
+		void SetupImageReader()
+		{
+			DisposeImageReader();
+
+			photoReader = ImageReader.NewInstance(640, 480, ImageFormatType.Jpeg, maxImages: 1);
+
+			var readerListener = new ImageAvailableListener();
+			readerListener.Photo += (_, bytes) =>
+			{
+				if (Element.SavePhotoToFile)
+					File.WriteAllBytes(ConstructMediaFilename(null, "jpg"), bytes);
+				Sound(MediaActionSoundType.ShutterClick);
+				OnPhoto(this, bytes);
+			};
+
+			photoReader.SetOnImageAvailableListener(readerListener, backgroundHandler);
+		}
+
+		void SetupMediaRecorder(Surface previewSurface)
+		{
+			DisposeMediaRecorder();
+
+			mediaRecorder = new MediaRecorder();
+			mediaRecorder.SetPreviewDisplay(previewSurface);
+			if (audioPermissionsGranted)
+				mediaRecorder.SetAudioSource(AudioSource.Camcorder);
+			mediaRecorder.SetVideoSource(AVideoSource.Surface);
+
+			var profile = GetCamcoderProfile();
+			if (profile != null)
+			{
```

Comment:
I thought I saw somewhere that the code style used is no brackets if a single line?